core/src/main/java/com/google/zxing/common/ECIStringBuilder.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2022 ZXing authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.zxing.common;
+
+import com.google.zxing.FormatException;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Class that converts a sequence of ECIs and bytes into a string
+ *
+ * @author Alex Geller
+ */
+
+public final class ECIStringBuilder {
+  private StringBuilder currentBytes;
+  private StringBuilder result;
+  private Charset currentCharset = StandardCharsets.ISO_8859_1;
+
+  public ECIStringBuilder() {
+    currentBytes = new StringBuilder();
+  }
+  public ECIStringBuilder(int initialCapacity) {
+    currentBytes = new StringBuilder(initialCapacity);
+  }
+
+  /**
+   * appends {@code value} as a byte value
+   */
+  public void append(char value) {
+    currentBytes.append((char) (value & 0xff));
+  }
+
+  /**
+   * appends {@code value} as a byte value (not it's string representation)
+   */
+  public void append(byte value) {
+    currentBytes.append((char) (value & 0xff));
+  }
+
+  /**
+   * appends the characters in {@code value} as bytes values
+   */
+  public void append(String value) {
+    currentBytes.append(value);
+  }
+
+  /**
+   * append the string repesentation of {@code value} (short for {@code append("" + value)})
+   */
+  public void append(int value) {
+    append("" + value);
+  }
+
+  public void appendECI(int value) throws FormatException {
+    encodeCurrentBytesIfAny();
+    CharacterSetECI characterSetECI = CharacterSetECI.getCharacterSetECIByValue(value);
+    if (characterSetECI == null) {
+      throw FormatException.getFormatInstance(new RuntimeException("Unsupported ECI value " + value));
```

Comment:
Pass the message directly, no need for RuntimeException

Reply:
Are you sure? I see only these two public methods:
public static FormatException getFormatInstance()
public static FormatException getFormatInstance(Throwable cause)

Reply:
Ah right, just leave out the msg (not super important)